crypto/vault/vault.go

```diff
@@ -0,0 +1,158 @@
+// vault.go - cryptographic vault for mixnet client key material
+// Copyright (C) 2017  David Anthony Stainton
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// crypto vault
+package vault
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/pem"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	"github.com/magical/argon2"
+	"golang.org/x/crypto/nacl/secretbox"
+)
+
+const (
+	// argon2SaltSize is the salt size for use with argon2
+	argon2SaltSize = 8
+
+	// passphraseMinSize is the minimum allowed passphrase size
+	passphraseMinSize = 12
+
+	// secretboxNonceSize is the nonce size for NaCl SecretBox
+	secretboxNonceSize = 24
```

Comment:
you should be able to specify this as the combination of source overheads. e.g as in https://github.com/privacylab/talek/blob/master/libtalek/topic.go#L33

Reply:
no because i'm using this value to generate the nonce of appropriate size whereas PublishingOverhead is different